src/ngMock/angular-mocks.js

```diff
@@ -135,6 +140,52 @@ angular.mock.$Browser = function() {
   self.baseHref = function() {
     return this.$$baseHref;
   };
+
+  self.interval = function(fn, interval) {
+    self.interval.repeatFns.push({
+      nextTime:(self.interval.now + interval),
+      delay: interval,
+      fn: fn,
+      id: self.interval.nextRepeatId
+    });
+    self.interval.repeatFns.sort(function(a,b) { return a.nextTime - b.nextTime;});
+
+    return self.interval.nextRepeatId++;
+  };
+
+  self.interval.cancel = function(id) {
+    var fnIndex;
+
+    angular.forEach(self.interval.repeatFns, function(fn, index) {
+      if (fn.id === id) fnIndex = index;
+    });
+
+    if (fnIndex !== undefined) {
+      self.interval.repeatFns.splice(fnIndex, 1);
+      return true;
+    }
+
+    return false;
+  };
+
+  self.interval.flush = function(millis) {
+    self.interval.now += millis;
+    while (self.interval.repeatFns.length &&
+        self.interval.repeatFns[0].nextTime <= self.interval.now) {
+      var task = self.interval.repeatFns[0];
+      task.fn();
+      task.nextTime += task.delay;
+      self.interval.repeatFns.sort(function(a,b) { return a.nextTime - b.nextTime;});
+    }
+    return millis;
+  };
+
+  self.interval.repeatFns = [];
+  self.interval.nextRepeatId = 0;
+  self.interval.now = 0;
+
+  self.shutdown = angular.noop;
```

Comment:
I didn't mean about leaking. I meant if someone wants to test that their app is behaves properly when destroyed. E.g. something like:

```js
describe('my interval-based carousel', function () {
  it('should stop changing when the app is destroyed', inject(function (...) {
    // Expect to show image-1
    // Flush
    // Expect to show image-2
    // Shutdown app
    // Flush
    // Expect to still show image-2 (not image-3)
  }));
})
```

But I guess it's outside of the scope of an app to test that, so it's fine as it is.